server/src/main/java/org/jboss/as/server/suspend/SuspendController.java

```diff
@@ -75,20 +81,32 @@ public synchronized void suspend(long timeoutMillis) {
         for(OperationListener listener: new ArrayList<>(operationListeners)) {
             listener.suspendStarted();
         }
-        outstandingCount = activities.size();
-        if (outstandingCount == 0) {
+        groupsCount = activities.size();
```

Comment:
Shouldn't this be
```suggestion
        // Work out the number of ServerActivity implementations 
        activities.forEach((ranking,executionGroup) -> groupsCount =+ executionGroup.size());
```

Or, we could also add a `groupsCount++` in `registerActivity(ServerActivity)`

Reply:
I don't think so. This var is meant to track the number of groups, not the number of activities. When each *group* completes preSuspend work it signals preSuspendGroupCallBack. When preSuspendGroupCallBack get groupsCount signals, it proceeds with the 'suspend' phase.

I should probably rename the 'activities' field to 'activitiesByGroup'.

Reply:
Yep, I see it now. Sorry for the noise.